### PR TITLE
feat: use Kyberswap ks-setting configurations

### PR DESCRIPTION
**Pull request by namgold**

### CODE REVIEW

File: src/hooks/useKyberswapConfig.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  classicClient: ApolloClient<NormalizedCacheObject>
+  elasticClient: ApolloClient<NormalizedCacheObject>
+  provider: ethers.providers.JsonRpcProvider
+}
+
+const convertConfig = (
+  data: KyberswapConfigurationResponse['data'] | undefined,
+  defaultChainId: ChainId,
+): KyberswapConfig => {
+  const rpc = isEVM(defaultChainId)
```
Comment (nguyenhoaidanh) [inline src/hooks/useKyberswapConfig.ts:31]: these configs for evm only?

File: src/services/ksSetting.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    rpc: string
+    'block-subgraph': string
+    'classic-subgraph': string
+    'elastic-subgraph': string
+  }
+}
+
+type KyberswapGlobalConfigurationResponse = {
+  data: {
+    banner: {
```
Comment (nguyenhoaidanh) [inline src/services/ksSetting.ts:18]: how about array of banners

File: src/services/ksSetting.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      id: string
+      name: string
+      start: string
+      end: string
+      link: string
+      img: {
```
Comment (nguyenhoaidanh) [inline src/services/ksSetting.ts:24]: img { 
mobile,
tablet,
desktop,
}

FE can check isInmodal/media and use the correct one

File: src/services/ksSetting.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  endpoints: builder => ({
+    getKyberswapConfiguration: builder.query<KyberswapConfigurationResponse, { chainId: ChainId }>({
+      query: ({ chainId }) => ({
+        url: '/configurations/fetch',
+        params: {
+          serviceCode: `kyberswap-${chainId}`,
+        },
+      }),
+    }),
+    getKyberswapGlobalConfiguration: builder.query<KyberswapGlobalConfigurationResponse, undefined>({
```
Comment (nguyenhoaidanh) [inline src/services/ksSetting.ts:47]: undefined => void

File: src/hooks/useKyberswapConfig.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        }
+      })
+      const results = await Promise.all(fetches)
+      setAllKyberswapConfig(
+        results.reduce(
+          (acc, cur) => {
+            acc[cur.chainId] = cur.result
+            return acc
+          },
+          {} as {
```
Comment (nguyenhoaidanh) [inline src/hooks/useKyberswapConfig.ts:95]: reduce<{ [chainId in ChainId]: KyberswapConfig}>

File: src/hooks/useKyberswapConfig.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  }, [getKyberswapConfiguration])
+
+  const defaultConfig = useMemo(
+    () =>
+      SUPPORTED_NETWORKS.reduce(
+        (acc, cur) => {
+          acc[cur] = convertConfig(undefined, cur)
+          return acc
+        },
+        {} as {
```
Comment (nguyenhoaidanh) [inline src/hooks/useKyberswapConfig.ts:111]: reduce<{ [chainId in ChainId]: KyberswapConfig}>

File: src/services/ksSetting.ts (hunk 4 above)
Comment (namgold) [reply src/services/ksSetting.ts:47]: is there any different between them?

Comment (nguyenhoaidanh) [reply src/services/ksSetting.ts:47]: undefined is value
void is type

File: src/hooks/useKyberswapConfig.ts (hunk 5 above)
Comment (namgold) [reply src/hooks/useKyberswapConfig.ts:95]: Have you tried this? Can you provide full code for this reduce

File: src/services/ksSetting.ts (hunk 4 above)
Comment (namgold) [reply src/services/ksSetting.ts:47]: undefined can be used as type too right?

File: src/hooks/useKyberswapConfig.ts (hunk 1 above)
Comment (namgold) [reply src/hooks/useKyberswapConfig.ts:31]: rpc config is EVM only.

File: src/hooks/useKyberswapConfig.ts (hunk 5 above)
Comment (nguyenhoaidanh) [reply src/hooks/useKyberswapConfig.ts:95]: lol, dont think it can be null here.
it's ok with your code.

File: src/services/ksSetting.ts (hunk 4 above)
Comment (nguyenhoaidanh) [reply src/services/ksSetting.ts:47]: yes, It's is my suggestion, it's up to u

File: src/services/ksSetting.ts (hunk 3 above)
Comment (namgold) [reply src/services/ksSetting.ts:24]: Tin chuan chua anh

Comment (namgold) [reply src/services/ksSetting.ts:24]: How about some special case that might required some special screenwidth

Comment (namgold) [reply src/services/ksSetting.ts:24]: Maybe it is not neccessary. I will update by your suggestion

Comment (namgold) [reply src/services/ksSetting.ts:24]: Updated

File: src/services/ksSetting.ts (hunk 2 above)
Comment (namgold) [reply src/services/ksSetting.ts]: My mistake, updated

**Review by nguyenhoaidanh — APPROVED** [review]

**Review by nguyenhoaidanh — DISMISSED** [review]

**Review by nguyenhoaidanh — APPROVED** [review]

**Pull request merged**